platform/linux-generic/odp_ipsec.c

```diff
@@ -1908,6 +1916,41 @@ int odp_ipsec_out_inline(const odp_packet_t pkt_in[], int num_in,
 	return in_pkt;
 }
 
+static void
+ipsec_sa_err_stats_update(odp_ipsec_packet_result_t *result)
+{
+	ipsec_sa_t *sa;
+
+	ODP_ASSERT(result->sa != ODP_IPSEC_SA_INVALID);
+
+	if (odp_likely(ODP_IPSEC_OK == result->status.all))
```

Comment:
It would be better to check if result->status.error.all is zero as we do not count soft lifetime expirations.